superset/views/geocoding.py

```diff
@@ -0,0 +1,327 @@
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+# pylint: disable=C,R,W
+
+import simplejson as json
+from flask import request, Response
+from flask_appbuilder import expose
+from flask_appbuilder.security.decorators import has_access, has_access_api
+from flask_babel import gettext as __
+from sqlalchemy import Column, Float, text
+from sqlalchemy.engine import Connection, reflection
+
+import superset.models.core as models
+from superset import appbuilder, conf, db
+from superset.connectors.sqla.models import SqlaTable
+from superset.exceptions import (
+    SqlAddColumnException,
+    SqlSelectException,
+    SqlUpdateException,
+)
+from superset.utils.geocoding_utils import GeoCoder
+
+from .base import api, BaseSupersetView, json_error_response, json_success
+
+
+class Geocoder(BaseSupersetView):
+    """Geocoding methods and API!"""
+
+    # Variables for geocoding
+    interruptflag = False
+    coder = GeoCoder(conf)
+
+    @has_access
+    @expose("/geocoding")
+    def geocoding(self):
+        """
+        Render react view for geocoding
+        :return: Geocoding react view
+        """
+        bootstrap_data = {
+            "tables": self._get_editable_tables(),
+            "common": self.common_bootstrap_payload(),
+        }
+
+        if request.args.get("json") == "true":
+            return json_success(
+                json.dumps(bootstrap_data, default=lambda x: x.__dict__)
+            )
+
+        return self.render_template(
+            "superset/geocoding.html",
+            entry="geocoding",
+            standalone_mode=False,
+            title="Geocode Addresses",
+            bootstrap_data=json.dumps(bootstrap_data, default=lambda x: x.__dict__),
+        )
+
+    def _get_editable_tables(self):
+        """
+        Get tables which are allowed to create columns (allow dml on their database)
+        :return: list of table names
+        """
+        tables = []
+        for database in (
+            db.session.query(models.Database).filter_by(allow_dml=True).all()
+        ):
+            for table in (
+                db.session.query(SqlaTable).filter_by(database_id=database.id).all()
+            ):
+                tables.append(models.TableDto(table.id, table.name, table.database_id))
+        return tables
+
+    @api
+    @has_access_api
+    @expose("/geocoding/columns", methods=["POST"])
```

Comment:
Why is this a POST?

Reply:
because the table name is sent to backend (post)